### ISSUE TITLE
Datetime functions in tremor std namespace

### ISSUE THREAD

**Issue by anupdhml** [wayfair-tremor/tremor-runtime#396]
We need to cleanup the datetime functions and make them available as part of tremor-script `std` namespace here:
https://github.com/wayfair-tremor/tremor-runtime/tree/main/tremor-script/lib/std

Underlying rust implementation (in-use currently by [datetime extractor](https://docs.tremor.rs/tremor-script/extractors/datetime/)):
https://github.com/wayfair-tremor/tremor-runtime/blob/main/tremor-script/src/std_lib/datetime.rs

**Comment by Licenser** [comment]
Yes! We need to clean up that module, a lot of the code can probably be implemented directly in tremorscript and not in a rust-function too which is kind of nice :)

**Comment by mfelsche** [comment]
We definitely need functions for producing timestamp strings (there are two RFC formats for this), as e.g. the google bigquery connector requires timestamp strings.

**Comment by Licenser** [comment]
#1978 included some additiuonal asks
